system/jlib/jlzw.cpp

```diff
@@ -2043,7 +2054,8 @@ class CCompressedFile : public CInterface, implements ICompressedFileIO
     }
     virtual ~CCompressedFile()
     {
-        close();
+        if (!writeException)
```

Comment:
Better than it was before, but close() could still throw an exception - when the final flush occurs.  Really need to catch exceptions in this destructor.